bundles/framework/publisher2/view/form/GeneralInfoForm.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import { LabeledInput, Select, Option, Message, Label } from 'oskari-ui';
+import styled from 'styled-components';
+import { InfoIcon } from 'oskari-ui/components/icons';
+
+const FieldWithInfo = styled('div')`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+`;
+
+const BUNDLE_KEY = 'Publisher2';
+
+export const GeneralInfoForm = ({ languageChange, onChange, data }) => {
+
+    const [name, setName] = useState(data.name.value ? data.name.value : null);
+    const [domain, setDomain] = useState(data.domain.value ? data.domain.value : null);
+    const [language, setLanguage] = useState( data.language.value ? data.language.value : Oskari.getLang());
+
+    const languages = Oskari.getSupportedLanguages();
+
+    return (
+        <div>
+            <FieldWithInfo>
+                <LabeledInput
+                    type='text'
+                    label={<Message messageKey='BasicView.name.label' bundleKey={BUNDLE_KEY} />}
+                    name='name'
+                    value={name}
+                    mandatory={true}
+                    onChange={(e) => {
+                        setName(e.target.value)
+                        onChange(e)
```

Comment:
Let's not expose the event outside this component. Instead onChange() could receive an object that holds all of the field values. Something like: 
```
{
    name: 'asdf,
    domain: 'adsGsa",
    lang: 'fi'
}

Reply:
It might read better if we just had ^ this as "state" object for the component and field updates would call setState() and onChange() with the same value.